securedrop/source.py

```diff
@@ -151,12 +151,10 @@ def generate_unique_codename(num_words):
 
 @app.route('/generate', methods=('GET', 'POST'))
 def generate():
-    # Popping this key prevents errors when a logged in user returns to /generate.
-    # TODO: is this the best experience? A logged in user will be automatically
-    # logged out if they navigate to /generate by accident, which could be
-    # confusing. It might be better to instead redirect them to the lookup
-    # page, or inform them that they're logged in.
-    session.pop('logged_in', None)
+    if logged_in():
+        flash("You were redirected because you are already logged in. If you want\
```

Comment:
Personally, for long strings, I prefer to use Python's line continuation behavior, which merges multiple quoted strings together automatically. Like this:

```py
flash("You were redirected because you are already logged in. If you want"
      "to create a new account, you should log out first.", "notification")
```